docs/plugins/haveibeenpwned/plugin.js

```diff
@@ -0,0 +1,250 @@
+                        this.passEditView.$el.addClass('input--error');
+                        InputFx.shake(this.passEditView.$el);
+                    }
+                });
+            } else {
+                this.logger.info("check pwnd passwd passed...");
+                this.passEditView.$el.removeClass('input--error');
+            }    
+        },
+        error: (e, xhr) => {
+            let err = xhr.response && xhr.response.error || new Error('Network error');
+            this.logger.error('Pwned Password API error', 'GET', xhr.status, err);
+            err.status = xhr.status;
+        }
+    });
+};
+
+DetailsView.prototype.fieldChanged = function (e) {
+    //this.logger.info('field changed ' + new HIBPUtils().stringify(e));
+    detailsViewFieldChanged.apply(this, [e]);
+    if (e.field) {
+        if (e.field === '$Password' && this.checkPwnedPwd) {
+            if (this.passEditView.value) {
+                const pwd = this.passEditView.value.getText();
+                if (pwd.replace(/\s/, '') !== '' && !pwd.startsWith('{REF:')) {
+                    new HIBPUtils().sha1(pwd).then(hash => {
+                        this.checkPwdPwned(hash.toUpperCase());
+                    });
+                }
+            }
+        } else if (e.field === '$UserName' && this.checkPwnedName) {
+            this.checkNamePwned(e.val);
+        }
+    }
+};
+
+module.exports.getSettings = function () {
+    return [{
+        name: 'checkPwnedPwd',
+        label: 'Check passwords against HaveIBeenPwned list',
+        type: 'checkbox',
+        value: 'true'
+    }, {
+        name: 'checkPwnedName',
+        label: 'Check user ids against HaveIBeenPwned list',
+        type: 'checkbox',
+        value: 'true'
+    }, {
+        name: 'blockPwnedPwd',
+        label: 'Block pwned passwords if they are in HaveIBeenPwned list',
+        type: 'checkbox',
+        value: 'true'
+    }, {
+        name: 'blockPwnedName',
+        label: 'Block pwned names if they are in HaveIBeenPwned list',
+        type: 'checkbox',
+        value: 'true'
+    }];
+};
+
+module.exports.setSettings = function (changes) {
+    // apply changed settings in plugin logic
+    // this method will be called:
+    // 1. when any of settings fields is modified by user
+    // 2. after plugin startup, with saved values
+    // only changed settings will be passed
+
+    // example: { MyText: 'value', MySel: 'selected-value', MyCheckbox: true }
+    // info(JSON.stringify(changes));
+    s = ''
+    for (field in changes) {
+        ccfield = field.substr(0, 1).toLowerCase() + field.substring(1);
+        DetailsView.prototype[ccfield] = changes[field];
+        //s += ccfield + '=' + DetailsView.prototype[ccfield] + '; ';
+    }
+    DetailsView.prototype.checkPwnedOnSettingsChanged.apply(changes);
+    //alert(s);
+};
+
+module.exports.uninstall = function () {
+    fieldChanged = detailsViewFieldChanged;
```

Comment:
Please replace with `fieldChanged ` with `DetailsView.prototype.fieldChanged`, changing a variable won't restore the prototype method.

Reply:
done, was a cut n paste error